packages/test/test-end-to-end-tests/src/test/summarizeRestart.spec.ts

```diff
@@ -0,0 +1,203 @@
+					mockConfigProvider(settings),
+				);
+
+			await summarizeNow(summarizer);
+
+			const dataStore = await requestFluidObject<ITestDataObject>(container, "default");
+			dataStore._root.set("an", "op");
+
+			await provider.ensureSynchronized();
+			const container2 = summarizingContainer2 as any;
+			// Intentional hack to allow two summaries to be submitted at the same time
+
+			const summarizeResults = summarizer.summarizeOnDemand({ reason: "end-to-end test" });
+			container2._clientId = (summarizer2 as any).runtime.summarizerClientId;
+
+			const summarizeResults2 = summarizer2.summarizeOnDemand({ reason: "end-to-end test" });
+
+			await Promise.all([
+				summarizeResults.summarySubmitted,
+				summarizeResults2.summarySubmitted,
+			]);
+
+			await Promise.all([
+				summarizeResults.summaryOpBroadcasted,
+				summarizeResults2.summaryOpBroadcasted,
+			]);
+
+			await Promise.all([
+				summarizeResults.receivedSummaryAckOrNack,
+				summarizeResults2.receivedSummaryAckOrNack,
+			]);
+			const ack = await summarizeResults.receivedSummaryAckOrNack;
+			const nack = await summarizeResults2.receivedSummaryAckOrNack;
+
+			await provider.ensureSynchronized();
+
+			assert(ack.success, "Should be an ack");
+			assert(!nack.success, "Should be a nack");
+			assert(summarizingContainer2.closed, "Unknown acks should close the summarizer");
+			assert(!summarizingContainer.closed, "summarizer1 should not be closed");
+			assert(!container.closed, "Original container should not be closed");
+		},
+	);
+
+	itExpects(
+		"Closes the summarizing client instead of refreshing when loading from an older summary",
+		[
+			{
+				eventName: "fluid:telemetry:Container:ContainerClose",
+				error: "Restarting summarizer instead of refreshing",
+			},
+			{
+				eventName: "fluid:telemetry:SummarizerNode:refreshLatestSummary_cancel",
+				error: "Restarting summarizer instead of refreshing",
+			},
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:HandleLastSummaryAckError",
+				error: "Restarting summarizer instead of refreshing",
+			},
+		],
+		async () => {
+			const container = await createContainer();
+			const { container: summarizingContainer, summarizer } = await createSummarizer(
+				provider,
+				container,
+				undefined,
+				undefined,
+				mockConfigProvider(settings),
+			);
+
+			// s1
+			const { summaryVersion } = await summarizeNow(summarizer);
+
+			const dataStore = await requestFluidObject<ITestDataObject>(container, "default");
+			dataStore._root.set("an", "op");
+
+			await provider.ensureSynchronized();
+			// s2
+			await summarizeNow(summarizer);
+			summarizer.close();
+			summarizingContainer.close();
+
+			const { container: summarizingContainer2, summarizer: summarizer2 } =
+				await createSummarizer(
+					provider,
+					container,
+					summaryVersion,
+					undefined,
+					mockConfigProvider(settings),
+				);
+
+			await provider.ensureSynchronized();
+			await summarizer2.run("abc");
```

Comment:
Is this necessary? When the new summarizer catches up, it would close right?

Reply:
I had a long conversation with @NicholasCouri, and we determined that it's actually when summarizer2.run gets called, which doesn't get called because we created the summarizer with disabledHeuristics.

Reply:
That makes sense. Can you add a comment explaining that.

Reply:
The summarizer doesn't start actually processing the acks until we call RunningSummarizer.start(). In this weird case, the second summarizer should not start as it isn't the elected summarizer client (hence the hack), so it doesn't process summary acks.